druid/src/widget/sub_window.rs

```diff
@@ -0,0 +1,114 @@
+use crate::app::WindowConfig;
+use crate::command::sys::SUB_WINDOW_PARENT_TO_HOST;
+use crate::commands::SUB_WINDOW_HOST_TO_PARENT;
+use crate::{
+    BoxConstraints, Command, Data, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+    PaintCtx, Point, Rect, Size, SubWindowRequirement, UpdateCtx, Widget, WidgetExt, WidgetId,
+    WidgetPod, WindowId,
+};
+use std::ops::Deref;
+
+/// A widget currently meant to be used as the root of a sub window.
+/// It ignores its own AppData, but provides its child with data synchronised
+/// from the WidgetPod that created it.
+pub struct SubWindowHost<U, W: Widget<U>> {
```

Comment:
Does this need to be public? Would it make sense for us to just handle the wrapping of whatever widget we're given in a host, internally?

If we can manage that I think this starts to feel really clean from an ergonomics perspective.

Reply:
I moved it out of the widget directory and it can be private then. If it stays in the widget directory it needs to be pub(crate)